addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -156,7 +156,18 @@ def get_promo_code_success_message(self, delete=True):
             request.session.pop('successful_code')
         return code
 
+    def _is_to_delete_discount(self, line, set_qty):
+        self.ensure_one()
+        return set_qty == 0 and line.coupon_id and line.reward_id and line.reward_id.reward_type == 'discount'
+
     def _cart_update(self, *args, **kwargs):
+        product_id, set_qty = kwargs['product_id'], kwargs.get('set_qty')
+
+        # Check if it's required to delete the line of the discount code
+        line = self.order_line.filtered(lambda l: l.product_id.id == product_id)
+        if set_qty == 0 and line.coupon_id and line.reward_id and line.reward_id.reward_type == 'discount':
+            kwargs['line_id'] = line.id
```

Comment:
```suggestion
        # Check if it's required to delete the line of the discount code
        line = self.order_line.filtered(lambda l: l.product_id.id == product_id)
        if set_qty == 0 and line.coupon_id and line.reward_id and line.reward_id.reward_type == 'discount':
            kwargs['line_id'] = None
```
Is clearer then

Reply:
Sorry but I don't understand why to use `None` instead of `line.id`?

Reply:
Following your explanation, line.id is None, so better make it clear and that case that's what we are doing here.
![image](https://github.com/odoo/odoo/assets/69820391/8d4c8b1b-bcee-4a47-bc79-696365391ada)